### DISCUSSION TITLE
GitHub PR Comment Link To Failed Linter Step?

### DISCUSSION THREAD

**Discussion by MattMencel** [oxsecurity/megalinter#2555] [Q&A]
When there is an error, the PR comment links to the MegaLinter workflow Summary. 
e.g. `https://github.com/myorg/myrepo/actions/runs/8675309`

Is there a way to have that link in to the failed job?
e.g. `https://github.com/myorg/myrepo/actions/runs/8675309/jobs/FAILEDJOBNUM`

Or even to the exact failed step?
e.g. `https://github.com/myorg/myrepo/actions/runs/8675309/jobs/FAILEDJOBNUM#step:5:217`

It's not a big deal, just a usability improvement.

**Comment by nvuillam** [comment]
I think at the time I created the reporter I looked for such link... but was unable to find it :/
But Github Actions evolved since... if you find a way to find the link to the failed step from MegaLinter code, we'd be delighted to make or accept a PR :)

**Reply by MattMencel** [reply]
@nvuillam looks like we can get down to the job anyway.... I found this code in `megalinter/reporters/GithubCommentReporter.py`. Just add the `GITHUB_JOB` and update the URL.

```python
    def produce_report(self):
        # Post comment on GitHub pull request
        if config.get("GITHUB_TOKEN", "") != "":
            github_repo = config.get("GITHUB_REPOSITORY")
            github_server_url = config.get("GITHUB_SERVER_URL", self.github_server_url)
            github_api_url = config.get("GITHUB_API_URL", self.github_api_url)
            run_id = config.get("GITHUB_RUN_ID")
            job_id = config.get("GITHUB_JOB")
            sha = config.get("GITHUB_SHA")

            if config.get("CI_ACTION_RUN_URL", "") != "":
                action_run_url = config.get("CI_ACTION_RUN_URL", "")
            elif run_id is not None:
                action_run_url = (
                    f"{github_server_url}/{github_repo}/actions/runs/{run_id}/jobs/{job_id}"
                )
            else:
                action_run_url = ""
```

Getting to the actual step is a little trickier. The format of the URL with the step reference is...

```
#step:STEP_ID:STEP_LINE_NUM
```

It may be possible to output the line number for each item, e.g. `[markdownlint]`... see below. The simple answer would be to hit a line approximately around where the linters start so STEP 5 (it's always step 5 I think) and LINE 75-100?

`e.g. https://github.com/myorg/myrepo/actions/runs/8675309/jobs/FAILEDJOBNUM#step:5:100`

I asked ChatGPT about getting to the actual line number because I had no clue. It may have given me the answer, I'm just not sure how or if you'd want to implement it.

```yaml
- name: Example step
  run: |
    echo "Hello World"
    echo "::set-output name=line_number::$(grep -n 'Hello World' $GITHUB_WORKSPACE/README.md | cut -d: -f1)"
```

It seems the steps append their output to a `$GITHUB_WORKSPACE/README.md` file and that is what is presented in the UI? So grabbing the line number would be a matter of grepping for the right string for each linter and using the `cut -d: -f1` to grab the line number.

Matt

**Reply by Kurt-von-Laven** [reply]
Looks like that question was a bit much for ChatGPT. Steps don't append their outputs to the README. At present LLMs work best when you can get them to cite sources or explain their answers. I don't understand the value of linking to the step though given that when you navigate to a failed job, it opens to the correct step by default. I see how linking directly to the failed job saves a click. How should the case where multiple jobs failed be handled?
